x-pack/plugins/security_solution/server/endpoint/routes/actions/audit_log_handler.ts

```diff
@@ -5,44 +5,69 @@
  * 2.0.
  */
 
-import { TypeOf } from '@kbn/config-schema';
 import { RequestHandler } from 'kibana/server';
-import { AGENT_ACTIONS_INDEX } from '../../../../../fleet/common';
-import { EndpointActionLogRequestSchema } from '../../../../common/endpoint/schema/actions';
+import { AGENT_ACTIONS_INDEX, AGENT_ACTIONS_RESULTS_INDEX } from '../../../../../fleet/common';
+import {
+  EndpointActionLogRequestParams,
+  EndpointActionLogRequestQuery,
+} from '../../../../common/endpoint/schema/actions';
 
 import { SecuritySolutionRequestHandlerContext } from '../../../types';
 import { EndpointAppContext } from '../../types';
 
 export const actionsLogRequestHandler = (
   endpointContext: EndpointAppContext
 ): RequestHandler<
-  TypeOf<typeof EndpointActionLogRequestSchema.params>,
-  unknown,
+  EndpointActionLogRequestParams,
+  EndpointActionLogRequestQuery,
   unknown,
   SecuritySolutionRequestHandlerContext
 > => {
   const logger = endpointContext.logFactory.get('audit_log');
+
   return async (context, req, res) => {
+    const {
+      params: { agent_id: elasticAgentId },
+      query: { page, page_size: pageSize },
+    } = req;
+
+    const size = pageSize;
+    const from = page <= 1 ? 0 : page * pageSize - pageSize + 1;
+
+    const options = {
+      headers: {
+        'X-elastic-product-origin': 'fleet',
+      },
```

Comment:
you should likely add ignore: 404 to this. 

Since a lack of entries isn't really an error situation for a history-log api

Reply:
done 34c05fcbb72995c3515346b00776e893d6c474bd